agent/consul/leader_connect_test.go

```diff
@@ -1065,3 +1068,50 @@ h1IHCbxWsUT3AiARwj5/D/CUppy6BHIFkvcpOCQoVyo=
 		}
 	}
 }
+
+func TestLeader_MinimumTTLFail(t *testing.T) {
+	t.Parallel()
+
+	monkey.Patch((*ca.ConsulProvider).MinimumLeafTTL, func(*ca.ConsulProvider) time.Duration {
```

Comment:
I really don't want to introduce unsafe runtime monkey patching into tests. At the very least that's something we'd want to RFC and agree on with other teams in the company but I don't think it's a good move even if we did.

If there is no other existing way to test this we should refactor to make it testable - let us know if you need help figuring out an alternative way to test this without patching. Also though this PR is doing the wrong thing so let's fix that before we refactor!

Not to mention the LICENSE for the monkey patch dep is incompatible with anything we can accept (I think the author likely did that on purpose to discourage use in real code bases that care about licensing and legal issues).